Content.Server/GameObjects/Components/Items/rcdComponent.cs

```diff
@@ -0,0 +1,166 @@
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Transform;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Server.GameObjects.Components.Items
+{
+    [RegisterComponent]
+    public class RcdComponent : Component, IAfterAttack, IUse, IExamine
+    {
+#pragma warning disable 649
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly IMapManager _mapManager;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager;
+#pragma warning restore 649
+        public override string Name => "Rapid Construction Device";
+        public string _outputTile = "floor_steel";
+        String[] modes = Enum.GetNames(typeof(RCDmodes)); //Displayed modes for saying stuff like "you switch to floors mode"
+        private int mode = 0; //What mode are we on? Can be floors, walls, deconstruct.
+        private int ammo = 5; //How much "ammo" we have left. You can refille this with RCD ammo.
+
+        private enum RCDmodes //Enum to store the different mode states for clarity.
+        {
+            Floors,
+            Walls,
+            Deconstruct
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _outputTile, "output", "floor_steel");
+        }
+
+        /**
+         *
+         *    Method called when the RCD is clicked in-hand, this will swap the RCD's mode from "floors" to "walls".
+         *
+         */
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
+        {
+            SwapMode(eventArgs);
+            return false;
+        }
+
+
+        /**
+         *
+         * Un-used method which is inherited from super
+         *
+         */
+        public void Activate(ActivateEventArgs eventArgs)
+        {
+            return;
+        }
+
+        /**
+         *
+         * Method to allow the user to swap the mode of the RCD by clicking it in hand, the actual in hand clicking bit is done over on UseEntity()
+         *
+         * @param UseEntityEventArgs = The entity which triggered this method call, used to know where to play the "click" sound.
+         *
+         */
+
+        public void SwapMode(UseEntityEventArgs eventArgs)
+        {
+            _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/genhit.ogg", Owner);
+            this.mode = (this.mode < modes.Length-1) ? this.mode + 1 : 0; //Basic value rollover so that you can't get to an invalid mode
```

Comment:
pretty sure you could do something like this instead:
`mode = (mode++) % modes.Length;`